config/routes.rb

```diff
@@ -8,6 +8,7 @@
       devise_scope :user do
         resources :users, only: %i[update]
         resources :targets, only: %i[index create destroy]
+        resources :general_informations, only: %i[show], param: :key
```

Comment:
Just to let you know when u have just one element you can omit the array and use `only: :show`